Sources/AirRobeWidget/Views/OptInView/AirRobeOptInView.swift

```diff
@@ -76,37 +76,42 @@ final class AirRobeOptInView: UIView, NibLoadable {
         // Widget Border Style
         mainContainerView.addBorder(color: AirRobeBorderColor.cgColor, cornerRadius: 0)
 
+        let tapOnArrowImage = UITapGestureRecognizer(target: self, action:  #selector(onTapArrow))
+        arrowImageView.isUserInteractionEnabled = true
+        arrowImageView.addGestureRecognizer(tapOnArrowImage)
+
+        let tapOnTopContentContainer = UITapGestureRecognizer(target: self, action:  #selector(onTapArrow))
+        topContentContainer.addGestureRecognizer(tapOnTopContentContainer)
+
         // Initializing Static Texts & Links
         titleLabel.text = UserDefaults.standard.OptedIn ? AirRobeStrings.added : AirRobeStrings.add
         descriptionLabel.text = AirRobeStrings.description
         potentialValueLabel.text = AirRobeStrings.potentialValue
         potentialValueLoading.hidesWhenStopped = true
         potentialValueLoading.startAnimating()
 
-        guard let appConfig = configuration else {
-            #if DEBUG
-            print("Widget is not yet configured.")
-            #endif
-            return
-        }
-        guard let privacyLink = URL(string: appConfig.privacyPolicyURL) else {
-            #if DEBUG
-            print("Privacy policy url is not valid.")
-            #endif
-            return
-        }
         detailedDescriptionLabel.setLinkText(
             orgText: AirRobeStrings.detailedDescription,
             linkText: AirRobeStrings.learnMoreLinkText,
-            link: privacyLink,
+            link: AirRobeStrings.learnMoreLinkForPurpose,
             tapHandler: onTapLearnMore)
         detailedDescriptionLabel.isHidden = true
-        margin.isHidden = true
-        extraInfoLabel.setLinkText(
-            orgText: AirRobeStrings.extraInfo,
-            linkText: AirRobeStrings.extraLinkText,
-            link: privacyLink,
-            tapHandler: onTapExtraInfoLink)
+
+        if let privacyLink = URL(string: AirRobeShoppingDataModelInstance.shared.shoppingDataModel?.data.shop.privacyUrl ?? "") {
+            extraInfoContainer.isHidden = false
+            let extraInfo = AirRobeStrings.extraInfo.replacingOccurrences(of: AirRobeStrings.companyNameText, with: AirRobeShoppingDataModelInstance.shared.shoppingDataModel?.data.shop.name ?? "")
+            extraInfoLabel.setLinkText(
+                orgText: extraInfo,
+                linkText: AirRobeStrings.extraLinkText,
+                link: privacyLink,
+                tapHandler: onTapExtraInfoLink)
+        } else {
+            extraInfoContainer.isHidden = true
+            #if DEBUG
+            print("Privacy policy url is not valid.")
+            #endif
+        }
+
```

Comment:
I find this block to be somewhat hard to follow because it is mixing multiple concerns -- if there is a privacyLink, then show the extraInfoContainer and set its link text based on the extraInfo / extraLinkText / privacyLink, along with this tapHandler. Otherwise, hide the extra info container.

Seeing multiple concerns in one block of code makes me wonder if privacyLink has too much responsibility? For example, what if there is extraInfo, extraLinkText, but there is no privacyLink? Does the lack of a privacyLink mean the extraInfo / extraLinkText should be ignored?

I wonder if this could be broken out a bit into a few bits of logic each with a single purpose?

Reply:
Yes, for now, we are hiding that extraInfo text block("By opting in...") if privacy url is `nil`.